orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -830,26 +839,35 @@ IMPLEMENT_GRADIENT_BUILDER(GetReluGradient) {
 
 IMPLEMENT_GRADIENT_BUILDER(GetSqueezeGradient) {
   std::vector<NodeDef> result;
-  auto attributes = SrcNodeAttributes();
-  std::vector<int64_t> axes_values;
-  if (attributes.find("axes") != attributes.end()) {
-    axes_values = RetrieveValues<int64_t>(attributes.at("axes"));
-    result.push_back(
-        NodeDef("Unsqueeze",
-                {GO(0)},
-                {GI(0)},
-                {MakeAttribute("axes", axes_values)}));
-    // if axes attribute not provided for squeeze
-  } else {
-    result.push_back(
-        NodeDef("Shape",
-                {I(0)},
-                {IA("I0_shape")}));
+  size_t numInputs = GetSrcNodeInputSize();
+  if (numInputs == 1) {
```

Comment:
>numInputs [](start = 6, length = 9)

I think it's better to explicitly call out the opset version here
e.g.  if opset <=12

Reply:
same for other ops 

---
In reply to: [510499951](https://github.com/microsoft/onnxruntime/pull/5488#discussion_r510499951) [](ancestors = 510499951)

Reply:
changing. there are situations even for opset 13 where we need to check num inputs as the 'axes' input is optional for ReduceSum, Squeeze.